pages/profile/index.js

```diff
@@ -25,23 +25,52 @@ export default function CarDetails() {
     event.preventDefault();
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
+
     const newCar = { ...data, UserId: user.id };
     try {
       const response = await fetch(`api/userCars/${user.id}`, {
         method: "PUT",
         body: JSON.stringify(newCar),
         headers: { "Content-type": "application/json" },
       });
-      setUser({ ...user, car: data.VIN });
       const responseCar = await response.json();
+      setUser({ ...user, car: data.VIN });
       setActiveCar(responseCar);
       setIsEditing(false);
       router.reload();
     } catch (error) {
       console.error(error);
     }
   }
-
+  function handleUploadFile(event) {
+    event.preventDefault();
+    if (event.target.files[0].size > 10485760) {
+      alert("Your picture is too big, Max 10 MB!");
+    }
+  }
+  async function handleSubmitPicture(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    const response = await fetch(`/api/upload`, {
+      method: "POST",
+      body: formData,
+    });
+    const imageUrl = await response.json();
+    const newCar = { ...activeCar, ImageUrl: imageUrl };
+    try {
+      const response = await fetch(`api/userCars/${user.id}`, {
```

Comment:
You are doing two requests in a row here. You could just have one api endpoint instead that uploads the picture and updates the car document in the database.

Reply:
Done